department-of-veterans-affairs/caseflow
# Pamatya/appeals 23401

Resolves [APPEALS-23401](https://jira.devops.va.gov/browse/APPEALS-23401)

# Description
1. Added a new component - powerOfAttorneyDecisionReview to disposition page. 
2. Added power of attorney method and other method required in decision_review_controller. 
3. Made change in the serializer. 
4. added feature test to test presence of power of attorney 
5. fixed Storybook 

## Acceptance Criteria
- [ ] Power of Attorney related to corresponding decision review should be displayed in between Header and decision section.
- [ ] Information provided about power of attorney should be correct. 
- [ ] if empty it should not contain any representative information. 


## Testing Plan

1 Login User : Any user which has access to VHA decision review. you can use - ACBAUERVVHA  or create user 
```
vha_user = FactoryBot.create(:user)
vha_org = BusinessLine.find_by_url( "vha")
vha_org.add_user(vha_user)
```
2. Create Higher level Review and Supplemental review 
```
veteran = FactoryBot.create(:veteran) 
FactoryBot.create(:higher_level_review,
             :with_vha_issue,
             :with_end_product_establishment,
             :create_business_line,
             benefit_type: "vha",
             veteran: veteran,
             number_of_claimants: 1)

sc = FactoryBot.create(:supplemental_claim, :processed, :with_vha_issue, :with_end_product_establishment, benefit_type: "vha", number_of_claimants: 1)
sc.create_business_line_tasks!
```
4. redirect to decision_reviews/vha after logging in. 
5.  on the table, click on the claimant name to view the decision review detail .
6.  Check if Appellant's Power Of attorney section is present. under veteran description and above decision section. 
7. Verify if the claimant detail match. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/888c24c0-2f1d-4f48-9422-b154027d79c8)|![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/c4be8871-04e9-40ca-a07e-fa83a3adc12e)
No POA | ![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/2233ddb9-14df-4c1c-bbe1-9e2337a87051)



## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
